srcpkgs/gnupg/template

```diff
@@ -18,10 +18,15 @@ maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
 license="GPL-3.0-or-later"
 homepage="https://www.gnupg.org/"
 distfiles="https://gnupg.org/ftp/gcrypt/gnupg/gnupg-${version}.tar.bz2"
-checksum=540b7a40e57da261fb10ef521a282e0021532a80fd023e75fb71757e8a4969ed
+checksum=1d79158dd01d992431dd2e3facb89fdac97127f89784ea2cb610c600fb0c1483
+alternatives="gpg:/usr/bin/gpg:/usr/bin/gpg2 gpgv:/usr/bin/gpgv:/usr/bin/gpgv2"
```

Comment:
No, this will break various program miserably:
```
$ gpgconf --check-programs                                            =79% =70%
gpg:OpenPGP:/usr/bin/gpg:1:1:
gpgsm:S/MIME:/usr/bin/gpgsm:1:1:
keyboxd:Public Keys:/usr/libexec/keyboxd:1:1:
gpg-agent:Private Keys:/usr/bin/gpg-agent:1:1:
scdaemon:Smartcards:/usr/libexec/scdaemon:1:1:
dirmngr:Network:/usr/bin/dirmngr:1:1:
pinentry:Passphrase Entry:/usr/bin/pinentry:1:1:
```

Reply:
Care to elaborate?

Reply:
Software that wants gnupg 2.x supposed to call `gpgconf --check-programs` and parse the output to detect which program provides which functionality. You're configure gpg 2.x to be built as `/usr/bin/gpg` but `/usr/bin/gpg` is not what it's claimed to (if the alternatives point to something else). At the very least configure gnupg with `--enable-gpg-is-gpg2`. I have no idea if any software expect gnupg functionalities, :shrug: 

Anyway, remember to remove the symlink in gnupg2 packages.

Reply:
IOW, the broken scenario is:

- users install both `gnupg` and `gnupg1`
- alternative set to `gnupg1`
- software call `gpgconf --check-programs` and think `/usr/bin/gpg` provides GnuPG v2 functionalities and process to call `/usr/bin/gpg` with GnuPG v2 flags.

Substitute `gnupg1` with any package provides alternatives for `/usr/bin/gpg`